src/components/AdminPanelLayout/UserListItem.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {FormEvent, useState} from 'react';
+import {Role, User} from '../../types/User';
+
+export interface ChangeUserRoleRequestSchema {
+  newRole: string;
+  userId: number;
+}
+
+interface Props {
+  user: User;
+  onRoleChange: (r: ChangeUserRoleRequestSchema) => void;
+}
+
+const roleOptions = [
+  {value: 'Guest' as Role, label: 'Guest'},
+  {value: 'Creator' as Role, label: 'Creator'},
+  {value: 'Admin' as Role, label: 'Admin'}
+];
+
+const UserListItem = (props: Props) => {
+
+  const [currentRole, setCurrentRole] = useState<Role>(props.user.role);
+
+  const handleRoleChange = (event: FormEvent<HTMLSelectElement>, user: User) => {
+    setCurrentRole(event.currentTarget.value as Role);
+    user.role = event.currentTarget.value as Role;
+    const newRequest: ChangeUserRoleRequestSchema =
+      {
+        userId: user.id,
+        newRole: event.currentTarget.value
+      };
+    props.onRoleChange(newRequest);
```

Comment:
inline this variable
```suggestion
    props.onRoleChange({
      userId: user.id,
      newRole: role
    });
```